### PR TITLE
added ansible tasks to deploy plugins and content service for aws cloud

### PR DESCRIPTION
**Pull request by G33tha**

### CODE REVIEW

File: ansible/inventory/env/group_vars/all.yml [hunk 1]
```diff
@@ -638,3 +638,12 @@ sunbird_report_service_url: "http://kong:8000/data/v1/report-service"
 
 # Datapipeline
 dp_play_http_secret_key: "mysecretdpplaysecretkey"
+
+# plugins deploy for aws,  if cloud storage type is AWS
+aws_access_key: {{aws_access_key | d('')}}
```
Comment (rjshrjndrn) [inline ansible/inventory/env/group_vars/all.yml:643]: keep it in the playbook itself.

File: ansible/roles/deploy-plugins/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+s3_bucket_name: {{aws_s3_bucket | d('')}}
```
Comment (rjshrjndrn) [inline ansible/roles/deploy-plugins/defaults/main.yml:1]: we don't have to override this vavriable, right ? we can use the same `aws_s3_bucket`

File: ansible/inventory/env/group_vars/all.yml (hunk 1 above)
Comment (rjshrjndrn) [reply ansible/inventory/env/group_vars/all.yml:643]: azure variables also should have a default empty, but both can't be empty.

File: ansible/roles/deploy-plugins/tasks/aws.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  shell: |
+    echo "{{sunbird_portal_preview_cdn_url}}"
+    sed -i 's|cdn_url|{{sunbird_portal_preview_cdn_url}}|g' "{{currentws}}"/ansible/preview/preview_cdn.html
+  when: sunbird_portal_preview_cdn_url is defined
+  tags:
+    - preview
+    
+- name: delete batch
+  shell: |
+    aws s3 rm s3://{{ aws_s3_bucket }}/{{ folder_name }}/ --recursive
```
Comment (rjshrjndrn) [inline ansible/roles/deploy-plugins/tasks/aws.yml:11]: in which task are we installing aws cli ?

File: ansible/roles/deploy-plugins/tasks/aws.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+- name: upload file
+  command: "aws s3 cp {{ source_file_name }} s3://{{ aws_s3_bucket }}/artefacts/content-player/content-player-{{ player_version_number }}.zip"
+  async: 3600
+  poll: 10  
+  tags:
+    - preview
+
+- name:  run s3_copy.sh
+  shell: "bash {{ s3_file_path }} {{ plugin_container_name }} {{ source_file }} {{ aws_region }}"
```
Comment (rjshrjndrn) [inline ansible/roles/deploy-plugins/tasks/aws.yml:40]: you don't have to define this variable as extra args, hard code the file path here, as this is always supposed to be cloud-specific script.

File: ansible/roles/deploy-plugins/tasks/azure.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+- name: upload file
+  command: "az storage blob upload --container-name {{ plugin_container_name }} --file {{ source_file_name }} --name artefacts/content-player/content-player-{{ player_version_number }}.zip"
+  async: 3600
+  poll: 10  
+  tags:
+    - preview
+
+- name:  run az_copy.sh
+  shell: "bash {{ az_file_path }} {{ plugin_container_name }} {{ source_file }}"
```
Comment (rjshrjndrn) [inline ansible/roles/deploy-plugins/tasks/azure.yml:49]: same as the above file path hardcoding

File: pipelines/deploy/CEPlugins/Jenkinsfile [hunk 6]
```diff
@@ -35,7 +35,7 @@ node() {
                        mv content-plugins ansible
                     """
                     ansiblePlaybook = "${currentWs}/ansible/deploy-plugins.yml"
-                    ansibleExtraArgs = "--tags plugins --extra-vars \" source_file=${currentWs}/ansible/content-plugins  az_file_path=${currentWs}/ansible/content-plugins/az_copy.sh\" --vault-password-file /var/lib/jenkins/secrets/vault-pass"  
+                    ansibleExtraArgs = "--tags plugins --extra-vars \" source_file=${currentWs}/ansible/content-plugins s3_file_path=${currentWs}/ansible/content-plugins/s3_copy.sh  az_file_path=${currentWs}/ansible/content-plugins/az_copy.sh\" --vault-password-file /var/lib/jenkins/secrets/vault-pass"
```
Comment (rjshrjndrn) [inline pipelines/deploy/CEPlugins/Jenkinsfile:38]: don't have to pass it as extra args. pass it in role itself

File: ansible/roles/deploy-plugins/tasks/aws.yml (hunk 3 above)
Comment (G33tha) [reply ansible/roles/deploy-plugins/tasks/aws.yml:11]: it has to be mannually installed on jenkins one time activity.

File: ansible/roles/deploy-plugins/defaults/main.yml (hunk 2 above)
Comment (G33tha) [reply ansible/roles/deploy-plugins/defaults/main.yml:1]: if there is no default value, we will get variable undefined error for azure @rjshrjndrn

File: ansible/roles/deploy-plugins/tasks/azure.yml (hunk 5 above)
Comment (G33tha) [reply ansible/roles/deploy-plugins/tasks/azure.yml:49]: file path cannot be hardcoded, it is in jenkins jobs workspace, so we are passing file path as extra args
ex: ${currentWs}/ansible/content-plugins
@rjshrjndrn

File: ansible/roles/stack-sunbird/defaults/main.yml [hunk 7]
```diff
@@ -664,3 +664,8 @@ report_liveness_readiness:
     periodSeconds: 10
     timeoutSeconds: 10
     failureThreshold: 5
+
+cloud_storage_type: azure
+aws_access_key: "{{aws_access_key | d('')}}"
```
Comment (rjshrjndrn) [inline ansible/roles/stack-sunbird/defaults/main.yml:669]: change the defaults

File: ansible/roles/deploy-plugins/tasks/azure.yml (hunk 5 above)
Comment (rjshrjndrn) [reply ansible/roles/deploy-plugins/tasks/azure.yml:49]:
```
    bash {{ az_file_path | d('content-plugins/test.sh') }} {{ plugin_container_name }} {{ source_file }}
```

File: ansible/roles/deploy-plugins/tasks/aws.yml (hunk 4 above)
Comment (rjshrjndrn) [reply ansible/roles/deploy-plugins/tasks/aws.yml:46]: check below comment

**Pull request merged**